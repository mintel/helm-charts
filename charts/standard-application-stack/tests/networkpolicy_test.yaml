suite: Test Network Policies
templates:
  - network-policy.yaml
release:
  namespace: test-namespace
tests:
  - it: Check ALB NetworkPolicy created if enabled
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].from
          value:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}

  - it: Check ALB NetworkPolicy created if enabled in extraIngresses
    set:
      global.clusterEnv: qa
      ingress.extraIngresses:
      - ingressNameOverride: $.data.appValues.global.name + '-api-gateway'
        defaultHost: 'qa-api.mintel.com'
        extraHosts: []
        pathBasedRouting:
        - targetService: test-service
          targetPort: 9999
          path: '/test/path'
          pathType: 'Prefix'
        alb:
          enabled: true
          scheme: 'internet-facing'
      # Disable default app networkpolicy
      networkPolicy.enabled: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].from
          value:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}

  - it: Check ALB NetworkPolicy created if internal alb selected
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
      ingress.alb.scheme: internal
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-internal-to-example-app
      - equal:
          path: spec.ingress[0].from
          value:
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_2}

  - it: Check ALB NetworkPolicy created if internal alb enabled in extraIngresses
    set:
      global.clusterEnv: qa
      ingress.extraIngresses:
      - ingressNameOverride: $.data.appValues.global.name + '-api-gateway'
        defaultHost: 'qa-api.mintel.com'
        extraHosts: []
        pathBasedRouting:
        - targetService: test-service
          targetPort: 9999
          path: '/test/path'
          pathType: 'Prefix'
        alb:
          enabled: true
          scheme: 'internal'
      # Disable default app networkpolicy
      networkPolicy.enabled: false
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-internal-to-example-app
      - equal:
          path: spec.ingress[0].from
          value:
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_2}

  - it: Check ALB NetworkPolicy ports set for default conatiner
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
      port: 8080
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].ports
          value:
            - port: 8080
              protocol: TCP

  - it: Check ALB NetworkPolicy ports set with oauth2proxy
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
      port: 8080
      oauthProxy.enabled: true
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].ports
          value:
            - port: 4180
              protocol: TCP

  - it: Check ALB NetworkPolicy ports set with oauth2proxy and healthcheck
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
      port: 8080
      oauthProxy.enabled: true
      ingress.alb.healthcheck.port: 9999
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].ports
          value:
            - port: 4180
              protocol: TCP
            - port: 9999
              protocol: TCP

  - it: Check ALB NetworkPolicy ports are unique
    set:
      global.clusterEnv: qa
      ingress.enabled: true
      # Disable default app networkpolicy
      networkPolicy.enabled: false
      ingress.alb.enabled: true
      port: 4180
      oauthProxy.enabled: true
      ingress.alb.healthcheck.port: 4180
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - isKind:
          of: NetworkPolicy
      - equal:
          path: metadata.name
          value: allow-aws-alb-to-example-app
      - equal:
          path: spec.ingress[0].ports
          value:
            - port: 4180
              protocol: TCP
