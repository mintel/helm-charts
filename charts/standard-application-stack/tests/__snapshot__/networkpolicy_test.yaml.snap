Check ALB NetworkPolicy created if enabled:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 8000
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy created if enabled in extraIngresses:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 8000
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy created if internal alb enabled in extraIngresses:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-internal-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 8000
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy created if internal alb selected:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-internal-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PRIVATE_APP_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 8000
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy ports are unique:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 4180
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy ports set for default conatiner:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 8080
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy ports set with oauth2proxy:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 4180
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
Check ALB NetworkPolicy ports set with oauth2proxy and healthcheck:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: NetworkPolicy
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: example-app
        app.mintel.com/application: example-app
        app.mintel.com/component: example-app
        app.mintel.com/env: qa
        app.mintel.com/region: ${CLUSTER_REGION}
        name: example-app
      name: allow-aws-alb-to-example-app
      namespace: test-namespace
    spec:
      ingress:
        - from:
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_0}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_1}
            - ipBlock:
                cidr: ${AWS_PUBLIC_SUBNET_CIDR_BLOCKS_2}
          ports:
            - port: 4180
              protocol: TCP
            - port: 9999
              protocol: TCP
      podSelector:
        matchLabels:
          app.kubernetes.io/part-of: null
